umbra-js/src/classes/Umbra.ts

```diff
@@ -235,6 +256,119 @@ export class Umbra {
     return { tx, stealthKeyPair };
   }
 
+  async batchSend(signer: JsonRpcSigner | Wallet, batch: SendBatch[], overrides: SendOverrides = {}) {
+    let tokenSum = new Map();
+    let tokens: string[] = [];
+    for (let i = 0; i < batch.length; i++) {
+      await assertSupportedAddress(batch[i].address);
+      const token = batch[i].token;
+      const amount = batch[i].amount;
+      if (tokenSum.get(token) == undefined) {
+        tokenSum.set(token, amount);
+        tokens.push(token);
+      } else {
+        tokenSum.set(token, tokenSum.get(token).add(amount));
+      }
+    }
+
+    // If applicable, check that sender has sufficient token balance. ETH balance is checked on send. The isEth
+    // method also serves to validate the token input
+    for (let j = 0; j < tokens.length; j++) {
+      const token = tokens[j];
+      if (!isEth(token)) {
+        const tokenContract = new Contract(token, ERC20_ABI, signer) as ERC20;
+        const tokenBalance = await tokenContract.balanceOf(await signer.getAddress());
+        if (tokenBalance.lt(tokenSum.get(token))) {
+          const providedAmount = BigNumber.from(tokenSum.get(token)).toString();
+          const details = `Has ${tokenBalance.toString()} tokens, tried to send ${providedAmount} tokens.`;
+          throw new Error(`Insufficient balance to complete transfer. ${details}`);
+        }
+      }
+    }
+
+    const txSigner = this.getConnectedSigner(signer); // signer input validated
+
+    // Get toll amount from contract
+    const toll = await this.umbraContract.toll();
+
+    // Parse provided overrides
```

Comment:
This also seems like a good helper function to create to avoid duplicating code

Reply:
Added paseOverrides helper 20d4e6f4518e7236c5467c398a8186cb17b7f319